### PR TITLE
2.3.0 Release notes

### PR DESCRIPTION
**Pull request by dreamer-89**
Signed-off-by: Suraj Singh <surajrider@gmail.com>

### Description
This changes adds the release notes for 2.3.0 release. 

Post merge to 2.3, I plan to backport to `main` and `2.x` branch
 
### Issues Resolved
https://github.com/opensearch-project/job-scheduler/issues/226
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dreamer-89** [comment]
Can I get some reviews on this PR ?

**Review by joshpalis — APPROVED** [review]

**Comment by dreamer-89** [comment]
There is no backport_main label, will raise PR manually for backport to `main`.

**Comment by dreamer-89** [comment]
Waiting for one more review on this PR (merge requirement).

**Review by saratvemulapalli — APPROVED** [review]

**Pull request merged**
